x-pack/plugin/inference/src/yamlRestTest/resources/rest-api-spec/test/inference/30_semantic_text_inference.yml

```diff
@@ -538,14 +514,261 @@ setup:
         id: doc_1
         body:
           source_field: "a single source field provided"
-          inference_field: "inference test"
+          sparse_field: "inference test"
 
   # Every source field needed on bulk updates
   - do:
       bulk:
         body:
           - '{"update": {"_index": "test-copy-to-index", "_id": "doc_1"}}'
-          - '{"doc": {"source_field": "a single source field is kept as provided via bulk", "inference_field": "updated inference test" }}'
+          - '{"doc": {"source_field": "a single source field is kept as provided via bulk", "sparse_field": "updated inference test" }}'
 
   - match: { items.0.update.status: 400 }
-  - match: { items.0.update.error.reason: "Field [another_source_field] must be specified on an update request to calculate inference for field [inference_field]" }
+  - match: { items.0.update.error.reason: "Field [another_source_field] must be specified on an update request to calculate inference for field [sparse_field]" }
+
+
+---
+"Update works for now - but will be unsupported later to avoid dealing with missing semantic_text fields content or copy_to fields":
```

Comment:
I thought we were going to support update if all source fields were provided, just like we do for bulk. Am I missing something?

Reply:
The problem is that we can't know that for sure on an update action, as it previously merges the existing doc with the changes in the update request. And it's treated as an indexing request instead of an update.

So we don't have a way of understanding whether all source fields have been provided, or they were already part of the doc.